mmdet/models/dense_heads/yolo_head.py

```diff
@@ -0,0 +1,581 @@
+        xy_use_logit (bool): Use log scale regression for bbox center
+            Default: False
+        balance_conf (bool): Whether to balance the confidence when calculating
+            loss. Default: False
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+            Default: dict(type='BN', requires_grad=True)
+        act_cfg (dict): Config dict for activation layer.
+            Default: dict(type='LeakyReLU', negative_slope=0.1).
+        train_cfg (dict): Training config of YOLOV3 head. Default: None.
+        test_cfg (dict): Testing config of YOLOV3 head. Default: None.
+    """
+
+    def __init__(self,
+                 num_classes,
+                 num_scales,
+                 num_anchors_per_scale,
+                 in_channels,
+                 out_channels,
+                 strides,
+                 anchor_base_sizes,
+                 ignore_thresh=0.5,
+                 one_hot_smoother=0.,
+                 xy_use_logit=False,
+                 balance_conf=False,
+                 norm_cfg=dict(type='BN', requires_grad=True),
+                 act_cfg=dict(type='LeakyReLU', negative_slope=0.1),
+                 train_cfg=None,
+                 test_cfg=None):
+        super(YOLOV3Head, self).__init__()
+        # Check params
+        assert (num_scales == len(in_channels) == len(out_channels) ==
+                len(strides) == len(anchor_base_sizes))
+        for anchor_base_size in anchor_base_sizes:
+            assert (len(anchor_base_size) == num_anchors_per_scale)
+            for anchor_size in anchor_base_size:
+                assert (len(anchor_size) == 2)
+
+        self.num_classes = num_classes
+        self.num_scales = num_scales
+        self.num_anchors_per_scale = num_anchors_per_scale
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.strides = strides
+        self.anchor_base_sizes = anchor_base_sizes
+
+        self.ignore_thresh = ignore_thresh
+        self.one_hot_smoother = one_hot_smoother
+        self.xy_use_logit = xy_use_logit
+        self.balance_conf = balance_conf
+
+        self.num_attrib = self.num_classes + 5
+        self.last_layer_dim = self.num_anchors_per_scale * self.num_attrib
+
+        cfg = dict(norm_cfg=norm_cfg, act_cfg=act_cfg)
+        self.convs_bridge = nn.ModuleList()
+        self.convs_final = nn.ModuleList()
+        for i_scale in range(self.num_scales):
+            in_c = self.in_channels[i_scale]
+            out_c = self.out_channels[i_scale]
+            conv_bridge = ConvModule(in_c, out_c, 3, padding=1, **cfg)
+            conv_final = nn.Conv2d(out_c, self.last_layer_dim, 1, bias=True)
+
+            self.convs_bridge.append(conv_bridge)
+            self.convs_final.append(conv_final)
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+    def init_weights(self):
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
```

Comment:
`ConvModule` is inited by itself.
Should we skip `ConvModule`s here?

Reply:
Yeah, I get your point. It makes sense, but these lines of code were just copied from `ResNet` (https://github.com/open-mmlab/mmdetection/blob/d613f216b2d9fa9dab3eaa4747f980cc386fb258/mmdet/models/backbones/resnet.py#L599)

So, I'm wondering if you are going to remove all these lines from mmdetection ?

Reply:
Is it the same as the official implementation of YOLOv3?

Reply:
Hi @hellock, 

I'm not sure, but I don't think different initializing methods will make huge difference. Btw, may I know why FCOS uses customized initial functions  https://github.com/open-mmlab/mmdetection/blob/d613f216b2d9fa9dab3eaa4747f980cc386fb258/mmdet/models/dense_heads/anchor_free_head.py#L138 .